iceoryx_utils/include/iceoryx_utils/internal/relocatable_pointer/pointer_repository.hpp

```diff
@@ -38,142 +42,71 @@ class PointerRepository
-        if (id > MAX_ID)
-        {
-            return false;
-        }
-        if (m_info[id].basePtr == nullptr)
-        {
-            m_info[id].basePtr = ptr;
-            m_info[id].endPtr = reinterpret_cast<ptr_t>(reinterpret_cast<uint64_t>(ptr) + size - 1u);
-            if (id > m_maxRegistered)
-            {
-                m_maxRegistered = id;
-            }
-            return true;
-        }
-        return false;
-    }
-
-    id_t registerPtr(const ptr_t ptr, uint64_t size = 0u)
-    {
-        for (id_t id = 1u; id <= MAX_ID; ++id)
-        {
-            if (m_info[id].basePtr == nullptr)
-            {
-                m_info[id].basePtr = ptr;
-                m_info[id].endPtr = reinterpret_cast<ptr_t>(reinterpret_cast<uint64_t>(ptr) + size - 1u);
-                if (id > m_maxRegistered)
-                {
-                    m_maxRegistered = id;
-                }
-                return id;
-            }
-        }
-
-        return INVALID_ID;
-    }
-
-    bool unregisterPtr(id_t id)
-    {
-        if (id <= MAX_ID && id >= MIN_ID)
-        {
-            if (m_info[id].basePtr != nullptr)
-            {
-                m_info[id].basePtr = nullptr;
+    /// @brief registers the start pointer of the segment in another application with a specific id
+    /// @param[in] id identifies the segment
+    /// @param[in] ptr is the start pointer of the segment
+    /// @param[in] size is the size of the segment
+    /// @return true if the registration was successful, otherwise false
+    bool registerPtr(id_t id, ptr_t ptr, uint64_t size) noexcept;
 
-                // do not search for next lower registered index but we could do it here
-                return true;
-            }
-        }
+    /// @brief registers the start pointer of a segment with a specific size
+    /// @param[in] ptr is the start pointer of the segment
+    /// @param[in] size is the size of the segment
+    /// @return the id that identifies the segment
+    id_t registerPtr(const ptr_t ptr, uint64_t size = 0U) noexcept;
 
-        return false;
-    }
+    /// @brief unregisters the id and also invalidates relative pointers using this id
+    /// @param[in] id is the id to be unregistered
+    /// @return true if successful, otherwise false
+    bool unregisterPtr(id_t id) noexcept;
 
-    void unregisterAll()
-    {
-        for (auto& info : m_info)
-        {
-            info.basePtr = nullptr;
-        }
-        m_maxRegistered = 0u;
-    }
-
-    ptr_t getBasePtr(id_t id)
-    {
-        if (id <= MAX_ID && id >= MIN_ID)
-        {
-            return m_info[id].basePtr;
-        }
+    /// @brief unregisters all ids and also invalidates all relative pointers
+    void unregisterAll() noexcept;
 
-        // for id 0 nullptr is returned, meaning we will later interpret a relative pointer
-        // by casting the offset into a pointer (i.e. we measure relative to 0)
+    /// @brief gets the base pointer associated with id
```

Comment:
Maybe confusing but base pointer and and starting address of a segment are used interchangebly in the code I wrote a long time ago :-) Maybe a comment would be helpful to clarify this...